deltacat/catalog/delegate.py

```diff
@@ -67,6 +69,19 @@ def read_table(
         **kwargs,
     )
 
+def create_scan_plan(
```

Comment:
Can we move this down from `catalog` to `storage`?

Everything in `catalog` is exposed to DeltaCAT end-users, but this should be serving compute framework integrations and the typical DeltaCAT end-user shouldn't ever need to know about it. 

As an example, the APIs in `catalog` should provide roughly equivalent functionality to the Iceberg Catalog API: https://github.com/apache/iceberg/blob/main/api/src/main/java/org/apache/iceberg/catalog/Catalog.java.

Reply:
Or, since we're requiring a `table` to always be specified anyways, attaching a `scan` method to `TableDefinition` may also make sense for now.

Reply:
Sounds good - I'll move the API to under `deltacat.catalog` as per suggestion.